src/main/java/io/vertx/core/file/impl/FileResolver.java

```diff
@@ -101,29 +100,32 @@ public FileResolver() {
   public FileResolver(FileSystemOptions fileSystemOptions) {
     this.enableCaching = fileSystemOptions.isFileCachingEnabled();
     this.enableCpResolving = fileSystemOptions.isClassPathResolvingEnabled();
-    this.fileCacheDir = fileSystemOptions.getFileCacheDir();
 
     String cwdOverride = System.getProperty("vertx.cwd");
     if (cwdOverride != null) {
       cwd = new File(cwdOverride).getAbsoluteFile();
     } else {
       cwd = null;
     }
-    if (this.enableCpResolving) {
-      setupCacheDir(UUID.randomUUID().toString());
-    }
+    cacheDir = setupCacheDir(fileSystemOptions.getFileCacheDir());
   }
 
   /**
    * Close this file resolver, this is a blocking operation.
    */
   public void close() throws IOException {
-    synchronized (this) {
-      if (shutdownHook != null) {
-        // May throw IllegalStateException if called from other shutdown hook so ignore that
-        try {
-          Runtime.getRuntime().removeShutdownHook(shutdownHook);
-        } catch (IllegalStateException ignore) {
+    // avoid monitor, if already disabled
+    if (shutdownHook != null) {
```

Comment:
you can remove this check as it is not synchronized and only check within the synchronized block.